keras_hub/src/models/yolo_v8/yolo_v8_detector.py

```diff
@@ -0,0 +1,604 @@
+
+    Args:
+        backbone: `keras.Model`, must implement the `pyramid_level_inputs`
+            property with keys "P3", "P4", and "P5" and layer names as values.
+            A sensible backbone to use is the `keras_hub.models.YOLOV8Backbone`.
+        num_classes: integer, the number of classes in your dataset excluding the
+            background class. Classes should be represented by integers in the
+            range [0, num_classes).
+        bounding_box_format: string, the format of bounding boxes of input dataset.
+            Refer
+            [to the keras.io docs](https://keras.io/api/keras_cv/bounding_box/formats/)
+            for more details on supported bounding box formats.
+        fpn_depth: integer, a specification of the depth of the CSP blocks in
+            the Feature Pyramid Network. This is usually 1, 2, or 3, depending
+            on the size of your YOLOV8Detector model. We recommend using 3 for
+            "yolo_v8_l_backbone" and "yolo_v8_xl_backbone". Defaults to 2.
+        label_encoder: (Optional)  A `YOLOV8LabelEncoder` that is
+            responsible for transforming input boxes into trainable labels for
+            YOLOV8Detector. If not provided, a default is provided.
+        prediction_decoder: (Optional)  A `keras.layers.Layer` that is
+            responsible for transforming YOLOV8 predictions into usable
+            bounding boxes. If not provided, a default is provided. The
+            default `prediction_decoder` layer is a
+            `keras_hub.layers.MultiClassNonMaxSuppression` layer, which uses
+            a Non-Max Suppression for box pruning.
+
+    Example:
+    ```python
+    images = tf.ones(shape=(1, 512, 512, 3))
+    labels = {
+        "boxes": tf.constant([
+            [
+                [0, 0, 100, 100],
+                [100, 100, 200, 200],
+                [300, 300, 100, 100],
+            ]
+        ], dtype=tf.float32),
+        "classes": tf.constant([[1, 1, 1]], dtype=tf.int64),
+    }
+
+    model = keras_hub.models.YOLOV8Detector(
+        num_classes=20,
+        bounding_box_format="xywh",
+        backbone=keras_hub.models.YOLOV8Backbone.from_preset(
+            "yolo_v8_m_backbone_coco"
+        ),
+        fpn_depth=2
+    )
+
+    # Evaluate model without box decoding and NMS
+    model(images)
+
+    # Prediction with box decoding and NMS
+    model.predict(images)
+
+    # Train model
+    model.compile(
+        classification_loss='binary_crossentropy',
+        box_loss='ciou',
+        optimizer=tf.optimizers.SGD(global_clipnorm=10.0),
+        jit_compile=False,
+    )
+    model.fit(images, labels)
+    ```
+    """  # noqa: E501
+
+    backbone_cls = YOLOV8Backbone
+
+    def __init__(
+        self,
+        backbone,
+        num_classes,
+        bounding_box_format,
+        fpn_depth=2,
+        preprocessor=None,
+        label_encoder=None,
+        prediction_decoder=None,
+        **kwargs,
+    ):
```

Comment:
restructure the code to define all teh layers first, functional model next and config last with these comments
=== Layers ===
.
.
=== Functional model ===
.
.
=== Config ===
.
.
example : https://github.com/keras-team/keras-hub/blob/master/keras_hub/src/models/bert/bert_backbone.py#L92

Reply:
Hi Divya, the current model applies multiple blocks of layers. That will imply that we would need to initialize many layers in the constructor. Moreover, the connections between those layers are not so straightforward as in Bert. What would you suggest? Shall we still initialize all layers in the layer block and connect them in the functional block?